external/vulkancts/modules/vulkan/conditional_rendering/vktConditionalClearAttachmentTests.cpp

```diff
@@ -0,0 +1,245 @@
+ * ------------------------
+ *
+ * Copyright (c) 2018 The Khronos Group Inc.
+ * Copyright (c) 2018 Danylo Piliaiev <danylo.piliaiev@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *//*!
+ * \file
+ * \brief Test for conditional rendering of vkCmdClearAttachments
+ *//*--------------------------------------------------------------------*/
+
+#include "vktConditionalClearAttachmentTests.hpp"
+#include "vktConditionalRenderingTestUtil.hpp"
+
+#include "vktTestCaseUtil.hpp"
+#include "vktDrawTestCaseUtil.hpp"
+
+#include "vktDrawBaseClass.hpp"
+
+#include "tcuTestLog.hpp"
+#include "tcuResource.hpp"
+#include "tcuImageCompare.hpp"
+#include "tcuTextureUtil.hpp"
+#include "tcuRGBA.hpp"
+
+#include "vkDefs.hpp"
+#include "vkCmdUtil.hpp"
+#include "vkTypeUtil.hpp"
+
+namespace vkt
+{
+namespace conditional
+{
+namespace
+{
+
+struct ConditionalTestSpec : public Draw::TestSpecBase
+{
+	ConditionalData	conditionalData;
+};
+
+class ConditionalClearAttachmentTest : public Draw::DrawTestsBaseClass
+{
+public:
+	typedef		ConditionalTestSpec	TestSpec;
+
+								ConditionalClearAttachmentTest	(Context &context, ConditionalTestSpec testSpec);
+
+	virtual		tcu::TestStatus iterate							(void);
+protected:
+	const ConditionalData			m_conditionalData;
+	de::SharedPtr<Draw::Buffer>		m_conditionalBuffer;
+
+	vk::Move<vk::VkCommandBuffer>	m_secondaryCmdBuffer;
+};
+
+ConditionalClearAttachmentTest::ConditionalClearAttachmentTest (Context &context, ConditionalTestSpec testSpec)
+	: Draw::DrawTestsBaseClass(context, testSpec.shaders[glu::SHADERTYPE_VERTEX], testSpec.shaders[glu::SHADERTYPE_FRAGMENT], vk::VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST)
+	, m_conditionalData(testSpec.conditionalData)
+{
+	checkConditionalRenderingCapabilities(context, m_conditionalData);
+
+	m_data.push_back(Draw::VertexElementData(tcu::Vec4(0.0f), tcu::Vec4(0.0f), 0));
+
+	initialize();
+
+	m_secondaryCmdBuffer = vk::allocateCommandBuffer(m_vk, m_context.getDevice(), *m_cmdPool, vk::VK_COMMAND_BUFFER_LEVEL_SECONDARY);
+};
+
+tcu::TestStatus ConditionalClearAttachmentTest::iterate (void)
+{
+	tcu::TestLog&		log		= m_context.getTestContext().getLog();
+	const vk::VkQueue	queue	= m_context.getUniversalQueue();
+	const vk::VkDevice	device	= m_context.getDevice();
+
+	const tcu::Vec4 clearColor	= tcu::RGBA::black().toVec();
+	const tcu::Vec4 drawColor	= tcu::RGBA::blue().toVec();
+
+	beginRenderPass(vk::VK_SUBPASS_CONTENTS_SECONDARY_COMMAND_BUFFERS);
```

Comment:
My previous comment about using `VK_SUBPASS_CONTENTS_SECONDARY_COMMAND_BUFFERS` is only for the case when using a secondary buffer for executing the commands; otherwise,  `VK_SUBPASS_CONTENTS_INLINE` should be used

Reply:
Check the other places too.

Reply:
Oh my... Yes, sorry, fixed now.